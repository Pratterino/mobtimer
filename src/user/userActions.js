import actionTypes from "./../actionTypes";
import {userImages} from "./userImages";
import _ from "lodash";

export const addUser = (name) => {
    const user = {
        name,
        active: false,
        image: _.sample(userImages),
    };

    return {
        type: actionTypes.ADD_USER,
        user,
    }
};

export const removeUser = (user) => {
    return {
        type: actionTypes.REMOVE_USER,
        user,
    }
};

export const nextUser = () => {
    return {
        type: actionTypes.NEXT_USER,
    }
};

export const toggleUser = (user) => {
    return {
        type: actionTypes.TOGGLE_USER,
        user,
    }
};
