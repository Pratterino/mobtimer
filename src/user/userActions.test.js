import React from 'react';
import { addUser, changeName, nextUser, removeUser, toggleUser, updateUserOrder } from './userActions';

describe('userActions', () => {
    describe('should match snapshots', () => {
        it('should match addUser', function () {
            let addUserResult = addUser('name');
            addUserResult.user.image = 'imageUrl';

            expect(addUserResult).toMatchSnapshot();
        });

        it('should match removeUser', function () {
            const user = {
                name: 'Bob Ducca',
            };
            expect(removeUser(user)).toMatchSnapshot();
        });

        it('should match nextUser', function () {
            expect(nextUser()).toMatchSnapshot();
        });

        describe('toggleUser', () => {
            it('when user is active', function () {
                const user = {
                    name: 'Bob Ducca',
                    active: true,
                };

                expect(toggleUser(user)).toMatchSnapshot();
            });

            it('when user is not active', function () {
                const user = {
                    name: 'Bob Ducca',
                    active: false,
                };

                expect(toggleUser(user)).toMatchSnapshot();
            });
        });

        it('should match changeName', function () {
            const user = {
                name: 'Bob Ducca',
            };

            expect(changeName(user, 'Scott Barnes')).toMatchSnapshot();
        });

        it('should match updateUserOrder', function () {
            const users = [
                {
                    name: 'Bob Ducca',
                },
                {
                    name: 'Bobby Duccas',
                },
            ];

            expect(updateUserOrder(users)).toMatchSnapshot();
        });
    });
});
