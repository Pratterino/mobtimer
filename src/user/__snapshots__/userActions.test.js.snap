// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`userActions should match snapshots should match addUser 1`] = `
Object {
  "type": "ADD_USER",
  "user": Object {
    "active": false,
    "image": "imageUrl",
    "name": "name",
  },
}
`;

exports[`userActions should match snapshots should match changeName 1`] = `
Object {
  "name": "Scott Barnes",
  "type": "CHANGE_USER_NAME",
  "user": Object {
    "name": "Bob Ducca",
  },
}
`;

exports[`userActions should match snapshots should match nextUser 1`] = `
Object {
  "type": "NEXT_USER",
}
`;

exports[`userActions should match snapshots should match removeUser 1`] = `
Object {
  "type": "REMOVE_USER",
  "user": Object {
    "name": "Bob Ducca",
  },
}
`;

exports[`userActions should match snapshots should match updateUserOrder 1`] = `
Object {
  "type": "UPDATE_USER_ORDER",
  "users": Array [
    Object {
      "name": "Bob Ducca",
    },
    Object {
      "name": "Bobby Duccas",
    },
  ],
}
`;

exports[`userActions should match snapshots toggleUser when user is active 1`] = `
Object {
  "type": "NOT_ALLOWED",
}
`;

exports[`userActions should match snapshots toggleUser when user is not active 1`] = `
Object {
  "type": "TOGGLE_USER",
  "user": Object {
    "active": false,
    "name": "Bob Ducca",
  },
}
`;
